rpadmin/api_migration.go

```diff
@@ -60,68 +62,39 @@ func (a *AdminAPI) ListMigrations(ctx context.Context) ([]MigrationState, error)
 
 // DeleteMigration deletes a migration by its ID.
 func (a *AdminAPI) DeleteMigration(ctx context.Context, id int) error {
-	return a.sendAny(ctx, http.MethodDelete, fmt.Sprintf("baseMigrationEndpoint%d", id), nil, nil)
-}
-
-// MigrationAction enum
-type MigrationAction int
-
-const (
-	// PrepareMigrationAction is the prepare migration action.
-	PrepareMigrationAction MigrationAction = iota
-
-	// ExecuteMigrationAction is the execute migration action.
-	ExecuteMigrationAction
-
-	// FinishMigrationAction is the finish migration action.
-	FinishMigrationAction
-
-	// CancelMigrationAction is the cancel migration action.
-	CancelMigrationAction
-)
-
-func (a MigrationAction) String() string {
-	switch a {
-	case PrepareMigrationAction:
-		return "prepare"
-	case ExecuteMigrationAction:
-		return "execute"
-	case FinishMigrationAction:
-		return "finish"
-	case CancelMigrationAction:
-		return "cancel"
-	default:
-		return ""
-	}
+	return a.sendAny(ctx, http.MethodDelete, fmt.Sprintf("%s%d", baseMigrationEndpoint, id), nil, nil)
 }
 
 // ExecuteMigration executes a specific action on a migration identified by its ID.
+// We set all migrations to auto_advance = true so there's generally no reason to call this
 func (a *AdminAPI) ExecuteMigration(ctx context.Context, id int, action MigrationAction) error {
-	if action < PrepareMigrationAction || action > CancelMigrationAction {
+	if action < MigrationActionPrepare || action > MigrationActionCancel {
 		return fmt.Errorf("invalid action: %s. Must be one of: prepare, execute, finish, cancel", action)
 	}
 	return a.sendAny(ctx, http.MethodPost, fmt.Sprintf("%s%d?action=%s", baseMigrationEndpoint, id, action), nil, nil)
 }
 
 // OutboundMigration represents an outbound migration request
 type OutboundMigration struct {
-	MigrationType  string   `json:"migration_type"`
-	Topics         []Topic  `json:"topics"`
-	ConsumerGroups []string `json:"consumer_groups"`
+	MigrationType  string            `json:"migration_type"`
+	Topics         []NamespacedTopic `json:"topics"`
+	ConsumerGroups []string          `json:"consumer_groups"`
+	AutoAdvance    bool              `json:"auto_advance"`
 }
 
 // InboundMigration represents an inbound migration configuration
 type InboundMigration struct {
 	MigrationType  string         `json:"migration_type"`
 	Topics         []InboundTopic `json:"topics"`
 	ConsumerGroups []string       `json:"consumer_groups"`
+	AutoAdvance    bool           `json:"auto_advance"`
 }
 
 // InboundTopic represents an inbound migration topic
 type InboundTopic struct {
-	SourceTopic Topic  `json:"source_topic"`
-	Alias       *Topic `json:"alias,omitempty"`
-	Location    string `json:"location,omitempty"`
+	SourceTopic NamespacedTopic  `json:"source_topic"`
+	Alias       *NamespacedTopic `json:"alias,omitempty"`
+	Location    string           `json:"location,omitempty"`
```

Comment:
I think `location` doesn't exist (anymore?). If seen this as well somewhere, but here's the schema I'm referring to